src/plugins/intel_gpu/tests/test_cases/non_zero_gpu_test.cpp

```diff
@@ -53,6 +53,82 @@ TEST(test_count_non_zero, 5d_fp16_1_3_2_1_2) {
     test_count_non_zero<FLOAT16>(layout{ov::PartialShape{1, 3, 2, 1, 2}, data_types::f16, format::bfzyx}, in_data);
 }
 
+TEST(test_count_non_zero, 2d_int32_1_256) {
+    const size_t len = 256;
+    std::vector<int32_t> in_data(len);
+    std::fill(in_data.begin(), in_data.end(), 1);
+
+    test_count_non_zero<int>(layout{ov::PartialShape{1, len}, data_types::i32, format::bfyx}, in_data);
+}
+
+TEST(test_count_non_zero, 2d_int32_1_513) {
+    const size_t len = 513;
+    std::vector<int32_t> in_data(len);
+    std::fill(in_data.begin(), in_data.end(), 1);
+
+    test_count_non_zero<int>(layout{ov::PartialShape{1, len}, data_types::i32, format::bfyx}, in_data);
+}
+
+TEST(test_count_non_zero, 2d_int32_1_1025) {
+    const size_t len = 1024;
+    std::vector<int32_t> in_data(len);
+    std::fill(in_data.begin(), in_data.end(), 1);
+
+    test_count_non_zero<int>(layout{ov::PartialShape{1, len}, data_types::i32, format::bfyx}, in_data);
+}
+
+class dyn_nonzero_count_net {
+public:
+    void create_network(cldnn::engine& engine, cldnn::topology& topology, ExecutionConfig& config) {
+        _network = std::make_shared<cldnn::network>(engine, topology, config);
+    }
+
+    std::map<primitive_id, network_output> execute(memory::ptr input_mem) {
+        _network->set_input_data("InputData", input_mem);
+        return _network->execute();
+    }
+
+private:
+    network::ptr _network;
+};
+
+TEST(test_count_non_zero, dynamic_test) {
+    auto& engine = get_test_engine();
+    auto in_dyn_layout = layout(ov::PartialShape::dynamic(2), data_types::i32, format::bfyx);
+
+    topology topology;
+    topology.add(input_layout("InputData", in_dyn_layout));
+    topology.add(count_nonzero("count_nonzero", input_info("InputData")));
+
+    ExecutionConfig config;
+    config.set_property(ov::intel_gpu::allow_new_shape_infer(true));
+
+    std::vector<size_t> input_shapes = {171, 531, 168, 169, 174, 172, 168, 167, 1169, 16, 677};
+    dyn_nonzero_count_net _test;
+    _test.create_network(engine, topology, config);
+
+    for (size_t idx = 0; idx < input_shapes.size(); idx++) {
+        const size_t input_length = input_shapes[idx];
+
+        ov::PartialShape p_shape  = { 1, static_cast<long int>(input_length) };
+        auto in_layout = layout(p_shape, data_types::i32, format::bfyx);
+        auto input_mem = engine.allocate_memory(in_layout);
+
+        std::vector<int32_t> in_data(input_length);
+        std::fill(in_data.begin(), in_data.end(), 1);
```

Comment:
This data is too simple to verify nonzero. 
* Could you fix this test to test more random data? 
e.g., 
1) fill random number (including floating point test) to the inputs 
2) add 0s to random positions 

* Also could you add test for blocked format too? 

*We dont need to validate such a many shapes. I think, some 2~3 shapes which are small than 256, larger than 1024, and nod divisable to power of two. We'll need to test 
1) more random numbers distributions (which contains definitly zeros) 
2) at least one float test and one integer test 
3) bfyx, bfzyx, bs_yx_fsv16  (here the feature shape should be > 16)

Reply:
I added bfyx, bfzyx, and bfzwyx format test but does not include bs_yx_fsv16 format because current nonzero_count does not support fsv16 format.